DSCResources/MSFT_xWebAppPool/MSFT_xWebAppPool.psm1

```diff
@@ -1,137 +1,1262 @@
+        [string]$shutdownTimeLimit = "00:01:30",
+
+        #Format 00:20:00
+        [string]$startupTimeLimit = "00:01:30",
+
+        [ValidateSet("true","false")]
+        [string]$pingingEnabled = "true",
+
+        #Format 00:20:00
+        [string]$pingInterval = "00:00:30",
+
+        #Format 00:20:00
+        [string]$pingResponseTime = "00:01:30",
+
+        [ValidateSet("true","false")]
+        [string]$disallowOverlappingRotation = "false",
 
-        [ValidateSet('Present','Absent')]
-        [System.String]
-        $Ensure  = 'Present',
+        [ValidateSet("true","false")]
+        [string]$disallowRotationOnConfigChange = "false",
 
-        [ValidateSet('Started','Stopped')]
-        [System.String]
-        $State = 'Started'
+        #format "Time, Memory, PrivateMemory"
+        [string]$logEventOnRecycle = "Time, Memory, PrivateMemory",
+
+        [string]$restartMemoryLimit = "0",
+
+        [string]$restartPrivateMemoryLimit = "0",
+
+        [string]$restartRequestsLimit = "0",
+
+        [string]$restartTimeLimit = "1.05:00:00",
+        
+        #Format 00:00:00 24hr clock and must have 00 for seconds
+        [string[]]$restartSchedule = @(""),
+
+        [ValidateSet("HttpLevel","TcpLevel")]
+        [string]$loadBalancerCapabilities = "HttpLevel",
+
+        [ValidateSet("true","false")]
+        [string]$orphanWorkerProcess = "false",
+
+        [string]$orphanActionExe = "",
+
+        [string]$orphanActionParams = "",
+
+        [ValidateSet("true","false")]
+        [string]$rapidFailProtection = "true",
+
+        #Format 00:20:00
+        [string]$rapidFailProtectionInterval = "00:05:00",
+
+        [string]$rapidFailProtectionMaxCrashes = "5",
+
+        [string]$autoShutdownExe = "",
+
+        [string]$autoShutdownParams = "",
+
+        [string]$cpuLimit = "0",
+
+        [ValidateSet("NoAction","KillW3wp","Throttle","ThrottleUnderLoad")]
+        [string]$cpuAction = "NoAction",
+
+        #Format 00:20:00
+        [string]$cpuResetInterval = "00:05:00",
+
+        [ValidateSet("true","false")]
+        [string]$cpuSmpAffinitized = "false",
+
+        [string]$cpuSmpProcessorAffinityMask = "4294967295",
+
+        [string]$cpuSmpProcessorAffinityMask2 = "4294967295"
     )
-    $WebAppPool = Get-TargetResource -Name $Name
+ 
+    $getTargetResourceResult = $null;
 
-    if($Ensure -eq 'Present')
+    if($Ensure -eq "Present")
     {
-        if($WebAppPool.Ensure -eq $Ensure -and $WebAppPool.State -eq $state)
+        #Remove Ensure from parameters as it is not needed to create new AppPool
+        $Result = $psboundparameters.Remove("Ensure");
+
+
+        # Check if WebAdministration module is present for IIS cmdlets
+        if(!(Get-Module -ListAvailable -Name WebAdministration))
         {
-            return $true
+            Throw "Please ensure that WebAdministration module is installed."
         }
+
+        #$AppPool = & $env:SystemRoot\system32\inetsrv\appcmd.exe list apppool $Name
```

Comment:
I'm not going to comment on ever line, but where you've commented out the appcmd.exe call could you please remove the line? It's not really needed at this point? :+1: